src/app/rosetta/lib/block.ml

```diff
@@ -0,0 +1,579 @@
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 1
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= None
+      ; failure_status=
+          Some
+            (`Applied
+              (Account_creation_fees_paid.By_fee_payer
+                 (Unsigned.UInt64.of_int 3_000)))
+      ; hash= "TXN_3b_HASH" }
+    ; { kind= `Create_account
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 1
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= None
+      ; failure_status= Some (`Applied Account_creation_fees_paid.By_no_one)
+      ; hash= "TXN_4_HASH" }
+    ; { kind= `Mint_tokens
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 10
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= Some (Unsigned.UInt64.of_int 30_000)
+      ; failure_status= Some (`Applied Account_creation_fees_paid.By_no_one)
+      ; hash= "TXN_5_HASH" } ]
+end
+
+module Internal_command_info = struct
+  module Kind = struct
+    type t = [`Coinbase | `Fee_transfer] [@@deriving eq]
+  end
+
+  type t =
+    { kind: Kind.t
+    ; receiver: [`Pk of string]
+    ; fee: Unsigned.UInt64.t
+    ; token: Unsigned.UInt64.t
+    ; hash: string }
+
+  let to_operations ~user_commands ~coinbase ~block_creator_pk (t : t) :
+      Operation.t list =
+    (* We choose to represent the dec-side of fee transfers from txns from the
+     * canonical user command that created them so we are able consistently
+     * produce more balance changing operations in the mempool or a block.
+     * *)
+    let plan : 'a Op.t list =
+      match t.kind with
+      | `Coinbase ->
+          [{Op.label= `Coinbase_inc; related_to= None}]
+      | `Fee_transfer ->
+          (* Detect if a fee transfer comes from a snark worker payment by
+         * failing to find the user command that caused it *)
+          if
+            List.find user_commands ~f:(fun (info : User_command_info.t) ->
+                Unsigned.UInt64.equal info.fee t.fee
+                && Unsigned.UInt64.equal info.token t.token
+                && [%eq: [`Pk of string]] info.receiver t.receiver )
+            |> Option.is_none
+          then
+            (* In this case we make a "synthetic" fee subtraction from block
+           * creator even though this is implicit in our blocks. We need this
+           * so the balance always sums to zero. *)
+            [ {Op.label= `Fee_payer_dec; related_to= None}
+            ; {Op.label= `Fee_receiver_inc; related_to= Some `Fee_payer_dec} ]
```

Comment:
delete the `Fee_payer_dec` here because you are already adding it for each user command. So basically for a fee transfer you'd have one op ```{Op.label= `Fee_receiver_inc; related_to= Some `Fee_payer_dec}``` whether it is for paying the coinbase receiver or the snark workers